src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -4,21 +4,42 @@
 using Microsoft.Win32.SafeHandles;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
-using System.Diagnostics.Contracts;
 using System.Runtime.InteropServices;
 using System.Security;
-using System.Threading;
-using System.Threading.Tasks;
 
 namespace System.IO.Pipes
 {
     public abstract partial class PipeStream : Stream
     {
+        private const string PipeDirectoryPath = "/tmp/";
+
+        internal static string GetPipePath(string serverName, string pipeName)
```

Comment:
I meant that mkfifo/open would fail because they'd end up looking for a file in a directory that didn't exist; there are certainly ways you could construct a path to get around that, though, e.g. with "../" to escape the tmp directory and then navigate elsewhere.  I've added the validation.